crowbar_framework/lib/execute.rb

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2013-2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "open3"
+
+class Execute
+  class << self
+    def bash(user, *params)
+      if user.nil?
+        return bash_as(false, *params)
+      elsif user == "root"
+        return bash_as(true, *params)
+      else
+        execution_uid = uid(user)
+        return false unless execution_uid
+
+        Process.fork do
+          Process.euid = execution_uid
+          spawn(*params)
+        end
+
+        pid, status = Process.wait2
+        return status.success?
+      end
+    end
+
+    def ruby(user, *params)
+      if user.nil?
+        return false
+      elsif user == "root"
+        return bash("root", "ruby", "-e", *params)
+      else
+        return ruby_as(user, *params)
+      end
+    end
+
+    protected
+
+    def bash_as(root, *params)
+      stdin, stdout, stderr, wait_thr = nil, nil, nil, nil
+      if root
+        stdin, stdout, stderr, wait_thr = Open3.popen3("sudo", "-i", *params)
+      else
+        stdin, stdout, stderr, wait_thr = Open3.popen3(*params)
+      end
+
+      return_value = wait_thr.value.success?
+      if return_value
+        return return_value
+      else
+        return return_value
+        Rails.logger.warn(stderr.read.strip)
+      end
+    end
+
+    def ruby_as(user, *params)
+      execution_uid = uid(user)
+      return false unless execution_uid
+
+      Process.fork do
+        Process.euid = execution_uid
+        eval(*params)
```

Comment:
You should disable this check for this specific line as we know what we are doing here.